src/utils/scales-utils.js

```diff
@@ -150,6 +150,32 @@ export function _getSmallestDistanceIndex(values, scaleObject) {
   return result;
 }
 
+/**
+ * This is a workaround for issue that ordinal scale
+ * does not have invert method implemented in d3-scale.
+ * @param {Object} Ordinal d3-scale object.
+ * @returns {void}
+ * @private
+ */
+
+function addInvertFunctionToOrdinalScaleObject(scale) {
+  if (scale.invert) return;
+
+  scale.invert = function invert(value) {
+    let domainIndex = 0;
+    const n = scale.domain().length;
+    const reverse = scale.range()[1] < scale.range()[0];
+    const start = scale.range()[reverse - 0];
+    const stop = scale.range()[1 - reverse];
+    
+    if (value < start + scale.padding() * scale.step()) domainIndex = 0;
+    else if (value > stop - scale.padding() * scale.step()) domainIndex = n - 1;
+    else domainIndex = Math.floor((value - start - scale.padding() * scale.step()) / scale.step());
+
+    return scale.domain()[domainIndex];
```

Comment:
rather than updating the domainIndex in each of the branches, why don't you just return the scale.domain()[VALUE] from one of the arms of the conditional?